hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/components/chartComponent/pieChartComponent.tsx

```
@@ -0,0 +1,138 @@
+            windowWidth: window.innerWidth
+        });
+    }
+
+    parseLabels = (label: string, charlen: number) => {
+        let res = "";
+        while(label.length > 0){
+            res += label.substring(0, charlen) + "<br>";
+            label = label.substring(charlen)
+        }
+        return res;
+    }
+
+    componentDidMount(): void {
+        window.addEventListener('resize', this.updateWindowSize);
+        this.props.plotData.forEach((data) => {
+            if (data && Object.keys(data).length !== 0){
+                if (data["labels"].some((label: string) => { return label.length > 35 })){
+                    this.setState({
+                        ...this.state,
+                        labelLengthOverflow: true
+                    })
+                }
+            }
+        });
+    }
+
+    componentWillUnmount(): void {
+        window.removeEventListener('resize', this.updateWindowSize);
+    }
+
+    render() {
+        const { windowHeight, windowWidth, labelLengthOverflow } = this.state;
+        let layoutProps: Partial<Plotly.Legend> = 
+        {
+            width: windowWidth * 0.8,
+            height: windowHeight - 100,
+            font: {
+                family: 'Roboto, sans-serif',
+                size: windowWidth * 0.008
+            },
+            showlegend: true,
+            legend: {
+                font: {
+                    size: windowWidth * 0.009
+                },
+                y: - windowHeight - 300,
+                yanchor: "bottom"
+            },
+            title: {
+                text: this.props.title,
+                font: {
+                    size: 20
+                }
+            },
+            margin: {
+                l: labelLengthOverflow ? windowWidth * 0.3 : 0,
+                r: (labelLengthOverflow && windowWidth < 1200) ? windowWidth * 0.2 : 0,
+                b: 100
+            },
+            paper_bgcolor: "#FFFFFF"
+        }
```

Are the hardcoded values here chosen during trial and error, or are there any references on how the values are chosen?

---

Hi @ivandika3 , sorry for the late reply, these values were chosen by trial and error and was mostly kind of taken as a binary search between the values.
So I kind of chose say between 1px and 5px, then if it didn't work I checked with 2.5px and like that for these values and then decided on the best fitting one for all the possible scenarios.

---

Thanks for the info.